From f73eb8f1dc84837c418f6cc147097b04698c0f86 Mon Sep 17 00:00:00 2001
From: Thierry FOURNIER <tfournier@exceliance.fr>
Date: Wed, 27 Nov 2013 15:30:55 +0100
Subject: [PATCH 150/179] MINOR: stick-table: use smp_expr_output_type() to
 retrieve the output type of a "struct sample_expr"

It's just a minor cleanup.
---
 src/stick_table.c | 14 +++++---------
 1 file changed, 5 insertions(+), 9 deletions(-)

diff --git a/src/stick_table.c b/src/stick_table.c
index b1d8827..0eac3e1 100644
--- a/src/stick_table.c
+++ b/src/stick_table.c
@@ -665,19 +665,15 @@ struct stktable_key *stktable_fetch_key(struct stktable *t, struct proxy *px, st
  */
 int stktable_compatible_sample(struct sample_expr *expr, unsigned long table_type)
 {
+	int out_type;
+
 	if (table_type >= STKTABLE_TYPES)
 		return 0;
 
-	if (LIST_ISEMPTY(&expr->conv_exprs)) {
-		if (!sample_to_key[expr->fetch->out_type][table_type])
-			return 0;
-	} else {
-		struct sample_conv_expr *conv_expr;
-		conv_expr = LIST_PREV(&expr->conv_exprs, typeof(conv_expr), list);
+	out_type = smp_expr_output_type(expr);
+	if (!sample_to_key[out_type][table_type])
+		return 0;
 
-		if (!sample_to_key[conv_expr->conv->out_type][table_type])
-			return 0;
-	}
 	return 1;
 }
 
-- 
1.8.1.5

